components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -5541,18 +5541,20 @@ private Attribute getSchemaAttributes(ExternalClaim scimClaim, LocalClaim mapped
                                           boolean isExtensionAttr) {
 
         String name = scimClaim.getClaimURI();
+        String claimDielectURI = scimClaim.getClaimDialectURI();
```

Comment:
Need to fix this issue properly.

For SCIM custom schema, attribute definition needs to be resolved from the properties defined in the mapped local claim.
as per the logic here: https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/blob/14c650b33f7509d1ba2d9635356e3960fde10dc4/components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCustomSchemaProcessor.java#L74

The custom schema's claims are not always simple singular value which has data type string. We can consider this is the case if the data type or multi-valued property is not defined in local claim properties.

Reply:
Resolved in https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/459/commits/b141ab8da6857ca63242aabfbe6b95cf10307319